docs/scope-and-objectives.md

```diff
@@ -0,0 +1,139 @@
+- __Management cluster__: The cluster where one or more Cluster API providers run, and where resources (e.g. Machines) are stored.
+- __Workload cluster__: A cluster whose lifecycle is managed by the Management cluster.
+- __Operating system__: A generically understood combination of a kernel and system-level userspace interface, such as Linux or Windows, as opposed to a particular distribution.
+- __Manage a cluster__: create, scale, upgrade, destroy.
+
+## Motivation
+
+Kubernetes has a common set of APIs (see the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)) to orchestrate containers regardless of deployment mechanism or cloud provider. Kubernetes also has APIs for handling some infrastructure, like load-balancers, ingress rules, or persistent volumes, but not for creating new machines.
+As a result, the deployment mechanisms that manage Kubernetes clusters each have unique APIs and implementations for how to handle lifecycle events like cluster creation or deletion, control plane upgrades, and node upgrades.
+
+### Goals
+
+- To manage the lifecycle (create, scale, upgrade, destroy) of Kubernetes-conformant clusters using a declarative API.
+- To work in different environments, both on-premises and in the cloud.
+- To define common operations, provide a default implementation, and provide the ability to swap out implementations for custom ones.
+- To reuse and integrate existing ecosystem components rather than duplicating their functionality (e.g. node-problem-detector, cluster autoscaler, SIG-Multi-cluster).
+- To provide a transition path for Kubernetes lifecycle products to adopt Cluster API incrementally. The result is a fragmentation of tooling, and we want to create a unified mechanism that gives users consistency around cluster management.
+
+### Non-goals
+
+- To add these APIs to Kubernetes core (kubernetes/kubernetes).
+   -  This API should live in a namespace outside the core and follow the best practices defined by api-reviewers, but is not subject to core-api constraints.
+- To manage the lifecycle of infrastructure unrelated to the running of Kubernetes-conformant clusters.
+- To force all Kubernetes lifecycle products (kops, kubespray, GKE, AKS, EKS, IKS etc.) to support or use these APIs.
+- To manage non-Cluster API provisioned Kubernetes-conformant clusters.
+- To manage a single cluster spanning multiple infrastructure providers.
+- To configure a machine at any time other than create or upgrade.
+- To duplicate functionality that exists or is coming to other tooling, e.g., updating kubelet configuration (c.f. dynamic kubelet configuration), or updating apiserver, controller-manager, scheduler configuration (c.f. component-config effort) after the cluster is deployed.
+
+## Requirements
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
```

Comment:
What parts of Cluster API are you thinking wouldn't be deployed on Kubernetes?

Reply:
Provider specific, mostly. But also any extension. As we have adopted the idea of swappable components, I'm wondering why this should be mandatory even if desirable.

Reply:
One reason this is here is to provide a mandate for the pivot workflow.

Reply:
So, I believe this would indeed be mandatory for anything that isn't a static hosted service. Otherwise like @ncdc mentioned we wouldn't be able to support pivoting Cluster API components.

That said, I'd be okay if we called out an exception for provider components that are externally hosted.